File: codechef/contests/APRIL16/AMAEXPER.cpp
#include <iostream>

using namespace std;

int main(void){
	int t, n, x, y, z, max = 100000;
	cin >> t;
	while (t--){
		cin >> n;
		// vector<int> a;
		for (int i = 0; i < n-1; i += 1){
			cin >> x >> y >> z;
			if (z < max) max = z;
		}
		cout << z << endl;
		for (int i = 0; i < n-1; i += 1){
			cout << "0" << endl;
		}
	}
}

File: codechef/contests/APRIL16/AMAEXPER.py


for _ in xrange(input()):
	n = input()
	a = []
	for i in xrange(n-1):
		x, y, z = map(int, raw_input().split())
		a.append(z)
	print max(a)
	for j in xrange(n-1):
		print 0